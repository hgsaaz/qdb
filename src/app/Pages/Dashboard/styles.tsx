import styled from 'styled-components/macro';
export const UserInfo = styled.p`
  padding: 0;
  margin: 0;
`;
