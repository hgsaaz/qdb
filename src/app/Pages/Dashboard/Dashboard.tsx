import React from 'react';
import { Tabs, Space, Divider } from 'antd';
import { Helmet, HelmetProvider } from 'react-helmet-async';
import { UserInfo } from './styles';
import { UserInfo as UserInterFace } from '../PageLayout/types';

const { TabPane } = Tabs;

interface Props {
  user: UserInterFace;
}

const Dashboard = Props => {
  const { user } = Props;
  const { address, company } = user || { address: null, company: null };

  return (
    <>
      <HelmetProvider>
        <Helmet titleTemplate="QDB Dashboard" defaultTitle="QDB Dashboard">
          <meta name="description" content="QDB application" />
        </Helmet>
      </HelmetProvider>
      <Tabs defaultActiveKey="1">
        <TabPane tab="OVERVIEW" key="1">
          <Space size="small" direction="vertical">
            <h3>Personal Details</h3>
            <UserInfo>Name: {user && user.name}</UserInfo>
            <UserInfo>Phone: {user && user.phone}</UserInfo>
            <UserInfo>Email: {user && user.email}</UserInfo>
            <UserInfo>Website: {user && user.website}</UserInfo>
            {address && (
              <>
                <Divider />
                <h3>Address</h3>
                <UserInfo>City: {address.city}</UserInfo>
                <UserInfo>Street: {address.street}</UserInfo>
                <UserInfo>Zipcode: {address.zipcode}</UserInfo>
              </>
            )}
            {company && (
              <>
                <Divider />
                <h3>Company</h3>
                <UserInfo>{user && user.company.name}</UserInfo>
                <UserInfo>{user && user.company.catchPhrase}</UserInfo>
              </>
            )}
          </Space>
        </TabPane>
        <TabPane tab="EXPENSE" key="2">
          <h3>Welcome {user && user.name}</h3>
          <Divider />
          <p>You can see Expense Reports here</p>
        </TabPane>
      </Tabs>
    </>
  );
};

export default Dashboard;
