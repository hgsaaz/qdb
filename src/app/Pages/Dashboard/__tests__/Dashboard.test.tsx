import React from 'react';
import { render, act } from '@testing-library/react';
import Dashboard from '../Dashboard';

const renderDashboard = (props: Parameters<typeof Dashboard>[number] = {}) =>
  render(<Dashboard {...props} />);
const email = 'test@test.com';
const name = 'test';
const testuser = {
  name: name,
  email: email,
};

describe('<Dashboard />', () => {
  it('should match the snapshot', () => {
    act(() => {
      const { container } = renderDashboard({
        user: testuser,
      });
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should have displayed props', () => {
    act(() => {
      const { container } = renderDashboard({
        user: testuser,
      });
      expect(container.querySelector('h3')).toMatchSnapshot();
    });
  });
});
