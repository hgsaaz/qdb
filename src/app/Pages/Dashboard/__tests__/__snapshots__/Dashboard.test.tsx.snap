// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should have displayed props 1`] = `
<h3>
  Personal Details
</h3>
`;

exports[`<Dashboard /> should have props displayed 1`] = `
<h3>
  Personal Details
</h3>
`;

exports[`<Dashboard /> should match the snapshot 1`] = `
.c0 {
  padding: 0;
  margin: 0;
}

<div
  class="ant-tabs ant-tabs-top"
>
  <div
    class="ant-tabs-nav"
    role="tablist"
  >
    <div
      class="ant-tabs-nav-wrap"
    >
      <div
        class="ant-tabs-nav-list"
        style="transform: translate(0px, 0px);"
      >
        <div
          class="ant-tabs-tab ant-tabs-tab-active"
        >
          <div
            aria-selected="true"
            class="ant-tabs-tab-btn"
            role="tab"
            tabindex="0"
          >
            OVERVIEW
          </div>
        </div>
        <div
          class="ant-tabs-tab"
        >
          <div
            aria-selected="false"
            class="ant-tabs-tab-btn"
            role="tab"
            tabindex="0"
          >
            EXPENSE
          </div>
        </div>
        <div
          class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
        />
      </div>
    </div>
    <div
      class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
    >
      <button
        aria-controls="null-more-popup"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-hidden="true"
        class="ant-tabs-nav-more"
        id="null-more"
        style="visibility: hidden; order: 1;"
        tabindex="-1"
        type="button"
      >
        <span
          aria-label="ellipsis"
          class="anticon anticon-ellipsis"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="ellipsis"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    class="ant-tabs-content-holder"
  >
    <div
      class="ant-tabs-content ant-tabs-content-top"
    >
      <div
        aria-hidden="false"
        class="ant-tabs-tabpane ant-tabs-tabpane-active"
        role="tabpanel"
        tabindex="0"
      >
        <div
          class="ant-space ant-space-vertical"
        >
          <div
            class="ant-space-item"
            style="margin-bottom: 8px;"
          >
            <h3>
              Personal Details
            </h3>
          </div>
          <div
            class="ant-space-item"
            style="margin-bottom: 8px;"
          >
            <p
              class="c0"
            >
              Name: 
              test
            </p>
          </div>
          <div
            class="ant-space-item"
            style="margin-bottom: 8px;"
          >
            <p
              class="c0"
            >
              Phone: 
            </p>
          </div>
          <div
            class="ant-space-item"
            style="margin-bottom: 8px;"
          >
            <p
              class="c0"
            >
              Email: 
              test@test.com
            </p>
          </div>
          <div
            class="ant-space-item"
          >
            <p
              class="c0"
            >
              Website: 
            </p>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        class="ant-tabs-tabpane"
        role="tabpanel"
        style="display: none;"
        tabindex="-1"
      />
    </div>
  </div>
</div>
`;
